org.eclipse.xtext/src/org/eclipse/xtext/validation/AbstractDeclarativeValidator.java

```diff
@@ -133,10 +141,15 @@ public void invoke(State state) {
 		}
 		
 		protected void handleInvocationTargetException(Throwable targetException, State state) {
-			// ignore GuardException, check is just not evaluated if guard is false
-			// ignore NullPointerException, as not having to check for NPEs all the time is a convenience feature
-			if (!(targetException instanceof GuardException) && !(targetException instanceof NullPointerException))
+			if (targetException instanceof NullPointerException) {
```

Comment:
Maybe it's better to offer a `handleExceptionDuringValidation(Throwable)` on the declarative validator and allow to override that one? This would offer more flexibility.

Reply:
Ping

Reply:
@kthoms ping

Reply:
Ok

Reply:
I guess the `swallowNPE` field becomes obsolete then. Shall I remove that?

Reply:
Yes, please.